libs/forecasting/src/lib/ForecastingDashboard/Controls/ForecastComparison.tsx

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Stack, Text } from "@fluentui/react";
+import {
+  ModelAssessmentContext,
+  defaultModelAssessmentContext,
+  BasicHighChart,
+  JointDataset,
+  orderByTime
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import { SeriesOptionsType } from "highcharts";
+import React from "react";
+
+import { forecastingDashboardStyles } from "../ForecastingDashboard.styles";
+
+import { getForecastPrediction } from "./getForecastPrediction";
+
+export class IForecastComparisonProps {}
+
+export interface IForecastComparisonState {
+  timeSeriesId?: number;
+  baselinePrediction?: number[];
+}
+
+const stackTokens = {
+  childrenGap: "l1"
+};
+
+export class ForecastComparison extends React.Component<
+  IForecastComparisonProps,
+  IForecastComparisonState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: IForecastComparisonProps) {
+    super(props);
+    this.state = {
+      baselinePrediction: undefined
+    };
+  }
+
+  public async componentDidMount(): Promise<void> {
+    const baselinePrediction = await this.getBaselineForecastPrediction();
+    if (baselinePrediction) {
+      this.setState({ baselinePrediction });
+    }
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    // Check if the time series was changed.
+    // In that case, we need to update our state accordingly.
+    const currentlySelectedTimeSeriesId =
+      this.context.baseErrorCohort.cohort.getCohortID();
+    if (currentlySelectedTimeSeriesId !== this.state.timeSeriesId) {
+      const baselinePrediction = await this.getBaselineForecastPrediction();
+      this.setState({
+        baselinePrediction,
+        timeSeriesId: currentlySelectedTimeSeriesId
+      });
+    }
+  }
+
+  public render(): React.ReactNode {
+    const classNames = forecastingDashboardStyles();
+
+    const indices = this.context.dataset.index;
```

Comment:
Since we are working on big data, will there be any issue with rendering big data here?

Reply:
Potentially! In short: we need to have a means to get the true Y, predicted Y, and any ad-hoc predictions based on transformations (that's the follow-on PR), so if there's a long forecast horizon (meaning we want not just the predictions for the next 5 points in time, but, say, thousands of them) then that could very well result in a "big data" case that should be handled by the backend. I will check with forecasting folks on typical forecast horizons and get back to you. I'll consider this a larger discussion that isn't blocking this PR if that's okay with you.

Reply:
Update: we expect on the order of tens and hundreds of predictions per time series, but not thousands because forecasting (intuitively) gets worse and worse the further you go. So I don't see this as a big issue in terms of data getting too big. I've made a note about input validation to warn users when they provide too big of a forecast horizon. This will be in my follow-up follow-up PR 🙂